prealloc.go

```diff
@@ -215,13 +226,32 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 					if !ok {
 						continue
 					}
-					if genD.Tok == token.VAR {
+					if genD.Tok == token.TYPE {
+						for _, spec := range genD.Specs {
+							tSpec, ok := spec.(*ast.TypeSpec)
+							if !ok {
+								continue
+							}
+
+							if _, ok := tSpec.Type.(*ast.ArrayType); ok {
+								arrayTypes = append(arrayTypes, tSpec.Name.Name)
```

Comment:
For safety, can we add a nil check on `tSpec.Name` here?
```Go
if tSpec.Name != nil {
    arrayTypes = append(arrayTypes, tSpec.Name.Name)
}
```

Reply:
I'm curious, could this ever be nil?

Reply:
I've wondered about this - it seems like the godoc for `go/ast` generally documents if the field can be nil. [TypeSpec](https://golang.org/pkg/go/ast/#TypeSpec)'s Name field is not documented as something that can potentially be nil, so perhaps we'd be okay here. I think for now: let's include it to be consistent with the rest of the codebase, but I might open an issue to circle back on these nil checks to see if they're truly needed.

Reply:
changed. :)